src/hotspot/os_cpu/bsd_aarch64/os_bsd_aarch64.hpp

```diff
@@ -41,30 +41,36 @@
   }
 
   #ifdef __APPLE__
-  enum WXMode {
-    WXUnknown,
-    WXWrite,
-    WXExec
-  };
 
-  // Enables write or execute access to writeable and executable pages.
-  // returns the previous state
-  static WXMode current_thread_change_wx(WXMode new_state);
+  class current_thread_wx {
```

Comment:
This violates the style guide for class names. It would be CurrentThreadWX - but ThreadWX seems sufficient to me.

Reply:
But os is okay? I just use this name for grouping.

Reply:
"os" is okay. Its historical. Its also more of a namespace really.

Reply:
but current_thread_wx would be too? Maybe I could change both to namespaces?

Reply:
But the style guide has no opinions on them...

Reply:
If/when the styleguide has an opinion on namespaces I would expect the same naming style to apply as for Classes.

Hotspot is full of historical quirks like "class os" I'm afraid.

Reply:
I changed it.